gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/processor/SynchronizationService.java

```diff
@@ -48,8 +53,49 @@ public <T> boolean checkSynchronization(final Class<T> entityClass, final T depl
                 boolean previousAccessibleState = f.isAccessible();
                 f.setAccessible(true);
                 try {
-                    requiredFieldsDeployedApi.add(f.get(deployedEntity));
-                    requiredFieldsApiToDeploy.add(f.get(entityToDeploy));
+                    if (f.get(deployedEntity) instanceof io.gravitee.definition.model.Proxy) {
+                        // FIXME - dirty check to ensure the Proxy -> Group -> Endpoint comparison is correct
+                        Endpoint deployedEntityProxyGroupEndpoint =
+                            ((Proxy) f.get(deployedEntity)).getGroups()
+                                .stream()
+                                .findFirst()
+                                .get()
+                                .getEndpoints()
+                                .stream()
+                                .findFirst()
+                                .get();
+                        Endpoint entityToDeployProxyGroupEndpoint =
+                            ((Proxy) f.get(entityToDeploy)).getGroups()
+                                .stream()
+                                .findFirst()
+                                .get()
+                                .getEndpoints()
+                                .stream()
+                                .findFirst()
+                                .get();
+
+                        if (!(deployedEntityProxyGroupEndpoint.equals(entityToDeployProxyGroupEndpoint))) {
+                            return false;
```

Comment:
This causes an issue I think. If I have two endpoints and edit the second one, the banner should show up. With this, it does not.